### PR TITLE
Replace townsfold gradle release plugin with researchgate release plugin.

### PR DESCRIPTION
**Pull request by Pscheidl**
This is how Sparkling Water invokes it:

`./gradlew -Prelease.useAutomaticVersion=true -Prelease.releaseVersion=${version} -Prelease.newVersion=${nextVersion} -PdoRelease release -x check`

https://github.com/h2oai/sparkling-water/blob/5ee475f0447386db83c7d00c69745630e35fa30b/ci/Jenkinsfile-release#L172

It's probably best to refer to the official documentation when it comes to CI usage: https://github.com/researchgate/gradle-release

Configuration:

https://github.com/h2oai/sparkling-water/blob/master/gradle/release.gradle

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -338,6 +338,8 @@ task buildH2oDevDist(type: Exec) {
     environment['BRANCH_NAME'] = branchName
     environment['BUILD_NUMBER'] = buildNumber
     environment['LAST_COMMIT_HASH'] = lastCommitHash
+    // For both master and release branch, return the version in gradle.properties to original version
+    environment['NEW_BUILD_VERSION'] = version
```
Comment (honzasterba) [inline build.gradle:342]: since this is actually the current version I would name it so, CURRENT_VERSION

**Comment by Pscheidl** [comment]
http://mr-0xc1:8080/view/H2O-3/job/h2o-3-release-pipeline/job/master/ will probably push "release" commits every night :(

File: gradle/release.gradle [hunk 2]
```diff
@@ -1,10 +1,11 @@
-apply plugin: 'release'
+apply plugin: 'net.researchgate.release'
 apply plugin: 'nexus-workflow'
 
 release {
-  tagPrefix = "release"
-  // Safe point - do releases only from release branch - can be deleted in future
-  //requireBranch = "testing"
+    failOnCommitNeeded = true
```
Comment (honzasterba) [inline gradle/release.gradle:5]: we also want to customize the commit messages, SW cod is good ispirations

**Comment by Pscheidl** [comment]
SW only has `tagTemplate = 'RELEASE-$version'` - we need to have jenkins-

File: gradle/release.gradle [hunk 3]
```diff
@@ -1,10 +1,15 @@
...
+    tagTemplate = 'jenkins-$version'
+    tagCommitMessage = "[H2O Release] :"
+    newVersionCommitMessage = "[New H2O Version] :"
```
Comment (honzasterba) [inline gradle/release.gradle:8]: this sould be more like "Reverting back to snaphost version "

**Comment by Pscheidl** [comment]
Done ! @honzasterba

**Pull request merged**